### ISSUE TITLE
Oracle GraalVM fails on AssertionError: no root node

### ISSUE THREAD

**Issue by Akirathan** [enso-org/enso#9477]
This happens when (dry-running) some benchmarks. Locally, this can be reproduced with:
```
sbt:std-benchmarks> benchOnly org.enso.benchmarks.generated.Map_Error_Benchmark_Slice_report_warning.Map_Id_All_Errors
```

Link to a corresponding failure on the CI: org.enso.benchmarks.generated.Map_Error_Benchmark_Slice_report_warning.Map_Id_All_Errors

The error is:
```
[error] java.lang.AssertionError: no root node for ArraySliceGen.TypesLibraryExports.Cached@4d523257
[error] with section: null
[error] with root nodes: null
[error] 	at org.enso.runtime/org.enso.interpreter.runtime.EnsoContext.reportSlowContextAccess(EnsoContext.java:243)
[error] 	at org.enso.runtime/org.enso.interpreter.runtime.EnsoContext.get(EnsoContext.java:218)
[error] 	at org.enso.runtime/org.enso.interpreter.runtime.data.vector.ArraySlice.getType(ArraySlice.java:185)
[error] 	at org.enso.runtime/org.enso.interpreter.runtime.data.vector.ArraySliceGen$TypesLibraryExports$Cached.getType(ArraySliceGen.java:101)
[error] 	at org.enso.runtime/org.enso.interpreter.runtime.library.dispatch.TypesLibraryGen$CachedDispatch.getType(TypesLibraryGen.java:529)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.InvokeMethodNodeGen.execute(InvokeMethodNodeGen.java:192)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.InvokeCallableNode.invokeDynamicSymbol(InvokeCallableNode.java:268)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.InvokeCallableNodeGen.execute(InvokeCallableNodeGen.java:149)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.ApplicationNode.executeGeneric(ApplicationNode.java:97)
[error] 	at org.enso.runtime/org.enso.interpreter.node.ClosureRootNode.execute(ClosureRootNode.java:85)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:746)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:669)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:586)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.callDirect(OptimizedCallTarget.java:535)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:94)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.thunk.ThunkExecutorNode.doCached(ThunkExecutorNode.java:69)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.thunk.ThunkExecutorNodeGen.executeThunk(ThunkExecutorNodeGen.java:123)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.argument.ArgumentSorterNode.executeArguments(ArgumentSorterNode.java:84)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.argument.ArgumentSorterNode.execute(ArgumentSorterNode.java:100)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.dispatch.InvokeFunctionNode.invokeCached(InvokeFunctionNode.java:107)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.dispatch.InvokeFunctionNodeGen.execute(InvokeFunctionNodeGen.java:81)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.InvokeCallableNode.invokeFunction(InvokeCallableNode.java:167)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.InvokeCallableNodeGen.execute(InvokeCallableNodeGen.java:125)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.ApplicationNode.executeGeneric(ApplicationNode.java:97)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.function.BlockNode.executeGeneric(BlockNode.java:54)
[error] 	at org.enso.runtime/org.enso.interpreter.node.ClosureRootNode.execute(ClosureRootNode.java:85)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:746)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:669)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:586)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.callDirect(OptimizedCallTarget.java:535)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:94)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.ExecuteCallNode.callDirect(ExecuteCallNode.java:94)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.ExecuteCallNodeGen.executeCall(ExecuteCallNodeGen.java:86)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.dispatch.SimpleCallOptimiserNode.executeDispatch(SimpleCallOptimiserNode.java:56)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.dispatch.CurryNode.doCall(CurryNode.java:161)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.dispatch.CurryNode.execute(CurryNode.java:107)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.dispatch.InvokeFunctionNode.invokeCached(InvokeFunctionNode.java:116)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.dispatch.InvokeFunctionNodeGen.execute(InvokeFunctionNodeGen.java:81)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.InvokeCallableNode.invokeFunction(InvokeCallableNode.java:167)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.InvokeCallableNodeGen.execute(InvokeCallableNodeGen.java:125)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.ApplicationNode.executeGeneric(ApplicationNode.java:97)
[error] 	at org.enso.runtime/org.enso.interpreter.node.scope.AssignmentNodeGen.executeGeneric_generic1(AssignmentNodeGen.java:78)
[error] 	at org.enso.runtime/org.enso.interpreter.node.scope.AssignmentNodeGen.executeGeneric(AssignmentNodeGen.java:55)
[error] 	at org.enso.runtime/org.enso.interpreter.node.scope.AssignmentNodeGen.executeVoid(AssignmentNodeGen.java:98)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.function.BlockNode.executeGeneric(BlockNode.java:52)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.function.BlockNode.executeGeneric(BlockNode.java:54)
[error] 	at org.enso.runtime/org.enso.interpreter.node.ClosureRootNode.execute(ClosureRootNode.java:85)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:746)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:669)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:602)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:586)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedCallTarget.callDirect(OptimizedCallTarget.java:535)
[error] 	at org.graalvm.truffle.runtime/com.oracle.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:94)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.ExecuteCallNode.callDirect(ExecuteCallNode.java:94)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.ExecuteCallNodeGen.executeCall(ExecuteCallNodeGen.java:86)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.dispatch.SimpleCallOptimiserNode.executeDispatch(SimpleCallOptimiserNode.java:56)
[error] 	at org.enso.runtime/org.enso.interpreter.node.callable.dispatch.CurryNode.doCall(CurryNode.java:161)
...
```

**Comment by Akirathan** [comment]
Another Oracle GraalVM - only error is reported in #9457

**Comment by JaroslavTulach** [comment]
I think [this line](https://github.com/enso-org/enso/commit/4332eab45a6c6f868f2854184ba0f9ba5412bb1e#diff-617035ec6aa85410ca0e9930d52bd9923006f2135b0c809a72cfc070b287123dR184) is reason for the `AssertionError`.

**Comment by JaroslavTulach** [comment]
Another [incarnation of the failure](https://github.com/enso-org/enso/actions/runs/8507628412/job/23299799958?pr=9584#step:8:2672).

**Comment by JaroslavTulach** [comment]
Will be solved by 4a398ecb7b to be integrated by  #9584

**Comment by JaroslavTulach** [comment]
Another fix will be provided by https://github.com/enso-org/enso/pull/9452/commits/380f5d2425a353ac1fa44351f31cf9140155fb29
